app/components/inputs/PathBrowseInput.js

```diff
@@ -0,0 +1,64 @@
+const electron = require("electron");
+const dialog = electron.remote.dialog;
+const mainWindow = electron.remote.getCurrentWindow();
+const ipc = electron.ipcRenderer;
+
+import PathInput from "./PathInput";
+import { PathButton } from "../buttons";
+
+let pathListener;
+
+@autobind
+class PathBrowseInput extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { path: "" };
+  }
+
+  componentDidMount() {
+
+    let self = this;
+    pathListener = function (event, data) {
+      self.setState({ path: data });
+      self.props.onChange(data);
+    };
+
+    ipc.on(this.props.id, pathListener);
```

Comment:
Instead of requiring that the user of `PathBrowseInput` pass an id prop, you can just generate a random one (generate a random number or a random string) and use it internally (since this has no effect for the parent component).

Reply:
good idea